solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java

```diff
@@ -124,7 +124,14 @@ private ClusterState fetchClusterState(
       throws SolrServerException, IOException, NotACollectionException {
     ModifiableSolrParams params = new ModifiableSolrParams();
     if (collection != null) {
+      if (log.isDebugEnabled()) {
+        log.debug("Making a call to Solr to fetch cluster state for collection: {}", collection);
+      }
       params.set("collection", collection);
+    } else {
+      if (log.isDebugEnabled()) {
+        log.debug("Making a call to Solr to fetch entire cluster state");
```

Comment:
From a performance perspective, I think requesting the entire state is unacceptable.  If we throw an UnsupportedOperationException here, what breaks?

Reply:
Fetching entire cluster state could be prevented from the server side - maybe as part of another PR/jira issue